examples/plotting/zoom_map_example.py

```diff
@@ -0,0 +1,45 @@
+import matplotlib.pyplot as plt
+from matplotlib import patches
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+###############################################################################
+# Define a region of interest.
+# A region of interest is a portion of an image that you want to operate on.
+# You can create many shape like circle, rectangle, polygon, ellipses etc.
+length = 500 * u.arcsec
+x0 = -50 * u.arcsec
+y0 = -30 * u.arcsec
+
+###############################################################################
+# We start with the sample data and
+# Create a SunPy Map and submap over the region of interest.
+smap = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+bottom_left = SkyCoord(x0 - length, y0 - length,
+                         frame = smap.coordinate_frame)
+top_right = SkyCoord(x0 + length, y0 + length,
+                         frame = smap.coordinate_frame)
+submap = smap.submap(bottom_left, top_right=top_right)
+
+fig = plt.figure(figsize=(16, 5))
+ax1 = fig.add_subplot(1, 2, 1, projection=smap)
+smap.plot(axes=ax1, clip_interval=(1, 99.9)*u.percent)
+
+# Draw a box on the image
+smap.draw_rectangle(bottom_left, height=length, width=length)
+
+# Create a second axis on the plot.
+ax2 = fig.add_subplot(1, 2, 2, projection=submap)
+
+submap.plot()
+
+# Add a overlay grid.
+submap.draw_grid(grid_spacing=20*u.deg)
+
+# name the zoomed plot.
+ax2.set_title('Zoomed View', fontsize=8)
```

Comment:
Can I ask why this example isn't as complex as the sample code in the original issue?

Reply:
Yes @nabobalis,
First forgive me as this is just my 3rd time contributing, i am still trying, it took me hours and days still i just reached here. That's why i make this as draft PR,
I need further guidance.

Also, taking about the code, here is the conversation between me and @wtbarnes 
Where he ask me a simple image rather than whole magnetogram thing. Also that code wasn't working.

Hey @wtbarnes Do we have to take a sample image (something like sample.AIA_171_IMAGE) and Plot the zoomed-in magnetogram on that particular image? That image part is not very clear to me. The image you posted up there, i am not finding anywhere in the documentation.

wtbarnes commented 10 days ago

Sorry, I should've been more clear in my original post. Yes, the gallery example would ideally use the sample data, e.g. the 171 image you mention.` It doesn't need to involve a magnetogram. The objective here is just to show how to plot a zoomed in view next to the original with the connector arrows.`

Reply:
Sure, working on examples easily brings the most amount of comments to keep up with.

But I don't understand why we can't take the code in that example, replace the image used and have that in this example as it was posted? I feel like that would be the easiest thing to do.

Reply:
I am not sure Nabil if i understand you. I think we can let this code as it is, as there were many plotting examples that are quite similar to this one(or can say based on same concept). Also i haven't seen any magnetogram related plotting in example section of codebase, so can we create a new issue saying that "example to plot a map using magnetogram"?

Reply:
The original issue had this image: https://user-images.githubusercontent.com/8676141/104660059-d776cb00-5693-11eb-84fd-89f94b690b60.png 

I am asking why we can't replicate that image for this example, by taking the code posted in the issue and fixing it so it works?

Reply:
If you suggest we can try. I was suggested to use sample  AIA_171_IMAGE

Reply:
I guess @wtbarnes and explain us better. I am getting confused. He put that image and magnetogram related codes tho he asked for simple zoom out plotting with sample sunpy image.
Can you help us @wtbarnes in figuring out what can be done.

Reply:
My intention behind adding that sample code in the original issue was do as suggested and just adapt it directly to the AIA 171 image as that is what is in our sample data set. Your example thus far is close to that, minus the connector lines. I've pasted the relevant bit of code for doing this in the main thread.